### ISSUE TITLE
Generated SARIF file is not valid SARIF

### ISSUE THREAD

**Issue by v-homsi** [cosmos/gosec#38]
### Summary 

Generated SARIF file is not considered valid as per [upload-sarif](https://github.com/github/codeql-action/tree/v2/upload-sarif) github action.

This is due to the duplicate ids of some of the rules generated in the SARIF file. 

### Steps to reproduce the behavior

To reproduce the error, you will need to run a github action similar to [this](https://github.com/evmos/evmos/blob/main/.github/workflows/security.yml). You can also check out the error in this [run](https://github.com/evmos/evmos/actions/runs/3049704173/jobs/4916039810). 

To reproduce the SARIF file, you can run:
```
git clone https://github.com/v-homsi/evmos.git
git checkout fix/gosec
go install github.com/cosmos/gosec/v2/cmd/gosec
gosec --no-fail -fmt sarif -out results.sarif ./...
```

### gosec version
Master branch

### Go version (output of 'go version')
go1.18.4

### Operating system / Environment
Ubuntu

### Expected behavior
Generated SARIF is valid and can be uploaded

### Actual behavior
Generated SARIF is not valid and cannot be uploaded

I cannot upload the sample file as the format is not supported by github. But here's a snippet of the SARIF file generated:

```
"tool": {
    "driver": {
        "name": "gosec",
        "informationUri": "https://github.com/securego/gosec/",
        "rules": [
            {
                "id": "G701 (CWE-)",
                "name": "Potential integer overflow by integer type conversion",
                "shortDescription": {
                    "text": "Potential integer overflow by integer type conversion"
                },
                "fullDescription": {
                    "text": "Potential integer overflow by integer type conversion"
                },
                "help": {
                    "text": "Potential integer overflow by integer type conversion\nSeverity: HIGH\nConfidence: MEDIUM\nCWE: "
                },
                "properties": {
                    "tags": [
                        "CWE-",
                        "HIGH"
                    ]
                }
            },
            {
                "id": "G701 (CWE-)",
                "name": "Potential integer overflow by integer type conversion",
                "shortDescription": {
                    "text": "Potential integer overflow by integer type conversion"
                },
                "fullDescription": {
                    "text": "Potential integer overflow by integer type conversion"
                },
                "help": {
                    "text": "Potential integer overflow by integer type conversion\nSeverity: HIGH\nConfidence: MEDIUM\nCWE: "
                },
                "properties": {
                    "tags": [
                        "CWE-",
                        "HIGH"
                    ]
                }
            },
            {
                "id": "G701 (CWE-)",
                "name": "Potential integer overflow by integer type conversion",
                "shortDescription": {
                    "text": "Potential integer overflow by integer type conversion"
                },
                "fullDescription": {
                    "text": "Potential integer overflow by integer type conversion"
                },
                "help": {
                    "text": "Potential integer overflow by integer type conversion\nSeverity: HIGH\nConfidence: MEDIUM\nCWE: "
                },
                "properties": {
                    "tags": [
                        "CWE-",
                        "HIGH"
                    ]
                }
            }
        ]
    }
},
```

**Comment by odeke-em** [comment]
Thank you for the report @v-homsi! Kindly tagging my colleague @kirbyquerby to take a look.

**Comment by kirbyquerby** [comment]
It looks like this was previously fixed in the upstream securego/gosec in https://github.com/securego/gosec/pull/565

I've made https://github.com/cosmos/gosec/pull/39 to pull in those changes, which should hopefully solve the issue. This repo is based on a pretty old version of gosec (almost 2 years old, I believe). Since this repository is primarily just additional rules on top of what securego/gosec provides, we would probably benefit a lot from figuring out a way to just add our rules to the latest version of gosec, rather than maintaining an entire fork. This will let us benefit from bug fixes, features, etc from upstream instead of having to discover and fix them separately ourselves.

**Comment by odeke-em** [comment]
Roger that @kirbyquerby and thank you for investigating!

**Comment by kirbyquerby** [comment]
@v-homsi this has hopefully been fixed by #39. I've released [v0.0.4](https://github.com/cosmos/gosec/releases/tag/v0.0.4).

I'm closing this as the action now succeeds in my fork: https://github.com/kirbyquerby/evmos/actions/runs/3063781539/jobs/4946236000

Feel free to reopen if something's still not working.

**Comment by v-homsi** [comment]
@kirbyquerby @odeke-em Works perfectly! Thanks for the help on this! Much appreciated :)
